src/Microsoft.ML.Core/Data/DataKind.cs

```diff
@@ -185,25 +185,25 @@ public static bool TryGetDataKind(this Type type, out DataKind kind)
             Contracts.CheckValueOrNull(type);
 
             // REVIEW: Make this more efficient. Should we have a global dictionary?
-            if (type == typeof(DvInt1) || type == typeof(sbyte) || type == typeof(sbyte?))
+            if (type == typeof(sbyte))
                 kind = DataKind.I1;
             else if (type == typeof(byte) || type == typeof(byte?))
                 kind = DataKind.U1;
-            else if (type == typeof(DvInt2)|| type== typeof(short) || type == typeof(short?))
+            else if (type == typeof(short))
                 kind = DataKind.I2;
-            else if (type == typeof(ushort)|| type == typeof(ushort?))
+            else if (type == typeof(ushort) || type == typeof(ushort?))
                 kind = DataKind.U2;
-            else if (type == typeof(DvInt4) || type == typeof(int)|| type == typeof(int?))
+            else if (type == typeof(int))
                 kind = DataKind.I4;
-            else if (type == typeof(uint)|| type == typeof(uint?))
+            else if (type == typeof(uint) || type == typeof(uint?))
                 kind = DataKind.U4;
-            else if (type == typeof(DvInt8) || type==typeof(long)|| type == typeof(long?))
+            else if (type == typeof(long))
                 kind = DataKind.I8;
-            else if (type == typeof(ulong)|| type == typeof(ulong?))
+            else if (type == typeof(ulong) || type == typeof(ulong?))
```

Comment:
> || type == typeof(ulong?)) [](start = 42, length = 27)

The use of nulltable types for the unsigned types seems wrong.

Reply:
Actually this entire function is wrong. This is meant to be an inverse to the function above. Didn't we already talk about this in a prior PR months ago?

---
In reply to: [211039836](https://github.com/dotnet/machinelearning/pull/683#discussion_r211039836) [](ancestors = 211039836)

Reply:
So: please remove all mentions of any types (whether nullable or not) that do not appear in the earlier function. However probably there's some code somewhere that relied on that change, try to figure out what it was, and fix it to use its own utility function.

---
In reply to: [211040069](https://github.com/dotnet/machinelearning/pull/683#discussion_r211040069) [](ancestors = 211040069,211039836)

Reply:
Hi @TomFinley, I believe this was added by @Ivanidzo4ka in #555 and you also approved that PR. I agree nullables here is wrong and I will remove them all.

Reply:
https://getyarn.io/yarn-clip/fc2be4a1-ca56-450b-8cbb-f771207b3df1

Anyway good. Not just the nullables but all but the single correct type. 😄